pkg/apis/workflow/v1alpha1/workflow_types.go

```diff
@@ -90,9 +91,28 @@ type ArtifactGCStrategy string
 const (
 	ArtifactGCOnWorkflowCompletion ArtifactGCStrategy = "OnWorkflowCompletion"
 	ArtifactGCOnWorkflowDeletion   ArtifactGCStrategy = "OnWorkflowDeletion"
-	ArtifactGCNever                ArtifactGCStrategy = ""
+	ArtifactGCNever                ArtifactGCStrategy = "Never"
+	ArtifactGCStrategyUndefined    ArtifactGCStrategy = ""
 )
```

Comment:
I needed to distinguish ArtifactGCNever from ArtifactGCStrategyUndefined, for the use case that somebody defines ArtifactGC on the Workflow level. On the artifact level they should be able to either not define it (ArtifactGCStrategyUndefined) or set it to ArtifactGCNever.

Reply:
yes